app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -1836,6 +1844,28 @@ class BrowserTabViewModel @Inject constructor(
         site?.consentCosmeticHide = isCosmetic
     }
 
+    fun onMaliciousSiteDetected(
```

Comment:
nit: I think it's best to rename this in a way that implies this handles the user actions, otherwise, it's a bit confusing to have `fun onMaliciousSiteDetected` and `fun onReceivedMaliciousSiteWarning`